### PR TITLE
Fix last animation keyframe bug

### PR DESCRIPTION
**Pull request by takahirox**
`writeOffset` for last keyframe in `KeyframeTrackPrototype.optimize()` is undefined so I added the declaration.

### CODE REVIEW

File: src/animation/KeyframeTrackPrototype.js [hunk 1]
```diff
@@ -336,12 +336,13 @@ KeyframeTrackPrototype = {
...
 
-		times[ writeIndex ++ ] = times[ lastIndex ];
+		times[ writeIndex ] = times[ lastIndex ];
 
-		for ( var readOffset = lastIndex * stride, j = 0; j !== stride; ++ j )
+		for ( var readOffset = lastIndex * stride, writeOffset = writeIndex * stride, j = 0; j !== stride; ++ j )
 
 			values[ writeOffset + j ] = values[ readOffset + j ];
 
+		++ writeIndex;
```
Comment (takahirox) [inline src/animation/KeyframeTrackPrototype.js]: BTW is `this.times.length > 1` guaranteed?
If not, I think here should be

    if ( lastIndex > 0 ) {

        times[ writeIndex ] = times[ lastIndex ];
  
        for ( var readOffset = lastIndex * stride, writeOffset = writeIndex * stride, j = 0; j !== stride; ++ j )
  
            values[ writeOffset + j ] = values[ readOffset + j ];

        ++ writeIndex;

    }

correct?

**Comment by tschw** [comment]
Looks good. Many thanks!

@takahirox 
> BTW is this.times.length > 1 guaranteed?

Good catch. The mixer can't handle empty tracks; it assumes each track provides at least one defined state. However, it might still be a good idea to allow `.optimize` to work in that case for editing. So yeah, the `if` would be a welcome addition (though checking for `lastIndex >= 0`, since it's offset by one from the length).

**Comment by mrdoob** [comment]
Thanks!

**Comment by takahirox** [comment]
Is there anything we have to do if `lastIndex` is 0?
`lastIndex` is 0 means `this.times.length` is 1.
In that case, that last keyframe operation just adds the same time/values.

ex: if this.times.length is 1

    before the last keyframe operation
    this.times: [0]
    this.values: [1, 2, 3] (stride is 3)

    after the last keyframe operation
    this.times: [0, 0]
    this.values: [1, 2, 3, 1, 2, 3]

Does this work as you intended?

**Comment by tschw** [comment]
@takahirox 
> [...] Is this result as you intended?

No - it's actually a defect. Thanks for clarifying the case.
Overlooked `writeIndex` starting at 1, I guess - so. yes, that `if` is needed for correctness.

**Pull request merged**